### PR TITLE
add instructions for triaging ICE issues

### PR DESCRIPTION
**Pull request by lolbinarycat**
based off of disscussion in
https://rust-lang.zulipchat.com/#narrow/channel/242269-t-release.2Ftriage/topic/Please.20make.20sure.20to.20label.20ICEs.20by.20source.20region

### CODE REVIEW

**Comment by lolbinarycat** [comment]
probably should also mention `-Z*` labels..

File: src/release/issue-triaging.md [hunk 1]
```diff
@@ -54,6 +54,16 @@ To see a list of all labels, check out the "labels" page next to the search bar
 [IRLO]: https://internals.rust-lang.org/
 [Discord]: https://discord.gg/rust-lang
 
+### ICE Triage
+For [Issues that have both I-ICE and needs-triage](https://github.com/rust-lang/rust/issues?q=is%3Aissue%20state%3Aopen%20label%3AI-ICE%20label%3Aneeds-triage)
+* If it an older (like latest stable) version of rust, ask for or check the latest nightly (potentially check for duplicates)
```
Comment (jieyouxu) [inline src/release/issue-triaging.md:59]: Nit: if it is

File: src/release/issue-triaging.md [hunk 2]
```diff
@@ -54,6 +54,16 @@ To see a list of all labels, check out the "labels" page next to the search bar
...
+* If it does not have a reproduction, comment asking for one and add S-needs-repro. if there isn't one for around a month it should generally be closed.
+* If the reproduction is not minimal, add `E-needs-mcve` or create a Minimal Complete and Verifyable Example yourself.
+* Add `A-*` labels based on the code that causes the issue (check backtraces!),
```
Comment (jieyouxu) [inline src/release/issue-triaging.md:62]: Suggestion: and maybe `T-compiler`/`T-types`/`T-rustdoc`/`WG-*`/`PG-*` as suitable?

File: src/release/issue-triaging.md (hunk 1 above)
Comment (jieyouxu) [inline src/release/issue-triaging.md:59]: Suggestion: do check for duplicates, but don't close them *too* eagerly, if it's not immediately clear they are due to the same root causes. Posting a cc to other potentially-duplicate issues is good.

File: src/release/issue-triaging.md [hunk 3]
```diff
@@ -54,6 +54,16 @@ To see a list of all labels, check out the "labels" page next to the search bar
 [IRLO]: https://internals.rust-lang.org/
 [Discord]: https://discord.gg/rust-lang
 
+### ICE Triage
+For [Issues that have both I-ICE and needs-triage](https://github.com/rust-lang/rust/issues?q=is%3Aissue%20state%3Aopen%20label%3AI-ICE%20label%3Aneeds-triage)
+* If it an older (like latest stable) version of rust, ask for or check the latest nightly (potentially check for duplicates)
+* If it does not have a reproduction, comment asking for one and add S-needs-repro. if there isn't one for around a month it should generally be closed.
+* If the reproduction is not minimal, add `E-needs-mcve` or create a Minimal Complete and Verifyable Example yourself.
```
Comment (jieyouxu) [inline src/release/issue-triaging.md:61]: Nit: verifiable, can also include `(MCVE)`

File: src/release/issue-triaging.md [hunk 4]
```diff
@@ -54,6 +54,16 @@ To see a list of all labels, check out the "labels" page next to the search bar
...
+* Add `A-*` labels based on the code that causes the issue (check backtraces!),
+  and the nature of the repro (eg. if the repro is a weird trait impl or the backtrace points to `rustc_trait_selection`, add `A-traits`)
+* Add `F-*`, `requires-*`, and `regression-*` labels as appropriate.
```
Comment (jieyouxu) [inline src/release/issue-triaging.md:64]: Remark: see above for more specific advice on those labels.

File: src/release/issue-triaging.md [hunk 5]
```diff
@@ -54,6 +54,16 @@ To see a list of all labels, check out the "labels" page next to the search bar
 [IRLO]: https://internals.rust-lang.org/
 [Discord]: https://discord.gg/rust-lang
 
+### ICE Triage
+For [Issues that have both I-ICE and needs-triage](https://github.com/rust-lang/rust/issues?q=is%3Aissue%20state%3Aopen%20label%3AI-ICE%20label%3Aneeds-triage)
```
Comment (jieyouxu) [inline src/release/issue-triaging.md:58]: Suggestion: first check if it's labelled correctly as `I-ICE`, some are more `I-crash` or `I-hang`, etc.

File: src/release/issue-triaging.md [hunk 6]
```diff
@@ -54,6 +54,16 @@ To see a list of all labels, check out the "labels" page next to the search bar
...
+* If it an older (like latest stable) version of rust, ask for or check the latest nightly (potentially check for duplicates)
+* If it does not have a reproduction, comment asking for one and add S-needs-repro. if there isn't one for around a month it should generally be closed.
```
Comment (jieyouxu) [inline src/release/issue-triaging.md:60]: Suggestion: check also if it's a regression:

- Especially double check if it's a regression-from-stable-to-{stable,beta,nightly}, or if unclear add `regression-untriaged`.
- Check if this needs prioritization.
- If it is a regression and there is a reproducer that's friendly to bisection within reasonable bisection step time, then tag `E-needs-bisection` if not already bisected and indicate to other people to aid with bisection, or run one yourself.
- If the bisection is able to pinpoint to a particular commit, double check if the commit roughly seems to be possibly correlated with the ICE itself -- sometimes the change can be more like "report more errors" which is not really the "root cause" of the ICE

File: src/release/issue-triaging.md (hunk 2 above)
Comment (jieyouxu) [inline src/release/issue-triaging.md:62]: Suggestion: also check if:

- ICE requires debug-assertions, add `requires-debug-assertions` as suitable.
- ICE requires incomplete features, add `requires-incomplete-features` as suitable.
- ICE requires unstable features, add `F-*` name as suitable.
- ICE requires internal features, close as won't-fix (in accordance with [MCP 620](https://github.com/rust-lang/compiler-team/issues/620)), but double check if it ICEs *without* needing internal features.

**Review by jieyouxu — APPROVED** [review]
Thanks, I have some more suggestions

File: src/release/issue-triaging.md (hunk 2 above)
Comment (lolbinarycat) [reply src/release/issue-triaging.md:62]: doesn't rustbot add T-compiler and I-Prioritize to all I-ICE issues?

File: src/release/issue-triaging.md (hunk 4 above)
Comment (lolbinarycat) [reply src/release/issue-triaging.md:64]: i'm confused as to why you left this comment telling me to look at the other comment on this same line of code.

File: src/release/issue-triaging.md (hunk 2 above)
Comment (lolbinarycat) [reply src/release/issue-triaging.md:62]: i already say to add F- and requires- labels?

i'll be honest, i'm finding this review a bit scattered and hard to follow, although i guess that's because i'm on the "Conversations" view.

File: src/release/issue-triaging.md (hunk 3 above)
Comment (lolbinarycat) [reply src/release/issue-triaging.md:61]: > can also include (MCVE)

not sure what you meant by this.

File: src/release/issue-triaging.md (hunk 4 above)
Comment (lolbinarycat) [reply src/release/issue-triaging.md:64]: ah, it's not the same line of code, GitHub is just being very confusing.

File: src/release/issue-triaging.md (hunk 2 above)
Comment (jieyouxu) [reply src/release/issue-triaging.md:62]: I'll post the feedback as one collectibe comment because the ui is extremely confusing for both you and me

**Comment by lolbinarycat** [comment]
oh, should also mention `O-*` labels.

File: src/release/issue-triaging.md (hunk 2 above)
Comment (jieyouxu) [reply src/release/issue-triaging.md:65]: Hm, I'm not sure actually, but what I had in mind is that people don't always consistency use the I-ICE template.

**Review by jieyouxu — APPROVED** [review]
Actually this looks good to me, we can always amend this in follow-up PRs.
